src/source.ts

```diff
@@ -45,6 +45,16 @@ export enum LoadingState {
   LOADED = 2,
 }
 
+/**
+ * Types of SourceOptions.
+ */
+export interface SourceOptions {
+  /**
+   * Indicates the player's offset time in seconds.
+   */
+  startOffset?: number;
+}
```

Comment:
The native side also defines a reference edge (start or end of the stream) this offset is applied to. 
Additionally, per default the reference edge for VoD stream is the start, while the reference edge for live streams is the end of the stream.
Further, having a negative number for the start of the stream does not make sense of course, because we can not start before 0.. similar with the end edge of the stream, where we can not start in the future of the live stream.

While I do not see a particular need to introduce the refrence edge config, I would ask you to update the documentation of this property a bit to reflect the behaviour a bit.
I am happy to answer further questions 🙂

Reply:
VoD:  start(currentTime=0) |--------------------| end (currentTime = duration)
Live: start(timeShift=maxTimeshift) |--------------------> end/live edge (timeShift=0)